### PR TITLE
Re-adds holopara injectors to uplink and ports "Majorly buffs standard holoparas"(with some changes)

### PR DESCRIPTION
**Pull request by gumner**
🆑 Gummer
add: Ports standard holopara buffs.
add: You can buy holopara injectors on uplinks once again.
/🆑

### CODE REVIEW

**Comment by GuyonBroadway** [comment]
OH SHIT NIGGER YOU FORGOT TO FORBID IT FROM NUKE OPS!

Actually part of me wants to leave it in just to see how funny it gets.

**Comment by tortellinitony** [comment]
dude it doesn't even compile

**Comment by gumner** [comment]
code/game/objects/effects/spawners/lootdrop.dm:137:error: /obj/item/weapon/storage/daki: undefined type path
code/game/objects/structures/crates_lockers/closets/secure/security.dm:32:error: /obj/item/weapon/storage/daki: undefined type path
code/modules/uplink/uplink_item.dm:1240:error: : expected 1 arg

**Comment by gumner** [comment]
great

**Comment by KittyKathy** [comment]
I'll try to help you.

**Comment by gumner** [comment]
thanks

File: code/modules/uplink/uplink_item.dm [hunk 1]
```diff
@@ -1230,7 +1237,7 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	while(crate_value)
 		var/category = pick(uplink_items)
 		var/item = pick(uplink_items[category])
-		var/datum/uplink_item/I = uplink_items[category][item]
+		var/datum/uplink_item/I = [category][item]
```
Comment (KittyKathy) [inline code/modules/uplink/uplink_item.dm]: Is there a reason for this change? It's causing the build to fail, you should change it back to `uplink_items[category][item]`.

Comment (gumner) [reply code/modules/uplink/uplink_item.dm]: mhm probally a missclick from me,thanks

**Comment by KittyKathy** [comment]
Oh also you need to open tgstation.dme with dream maker and find the files
code\game\objects\items\weapons\storage\dakis.dm
code\modules\mob\living\carbon\monkey\emote.dm
code\modules\surgery\gender_reassignment.dm
and click the little checkbox next to them so they get included I think.

**Comment by yobob591** [comment]
jojo ops

File: code/modules/uplink/uplink_item.dm [hunk 2]
```diff
@@ -558,6 +559,12 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	item = /obj/mecha/combat/gygax/dark/loaded
 	cost = 80
 
+/datum/uplink_item/dangerous/guardian
+	name = "Holoparasites"
+	desc = "Though capable of near sorcerous feats via use of hardlight holograms and nanomachines, they require an organic host as a home base and source of fuel."
+	item = /obj/item/weapon/storage/box/syndie_kit/guardian
+	cost = 20
+
```
Comment (KittyKathy) [inline code/modules/uplink/uplink_item.dm]: You can add this line `exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/gang)` after the cost to make it traitor gamemode only.

Comment (gumner) [reply code/modules/uplink/uplink_item.dm]: yeah i tried and it said something like unknown var

**Comment by JohnGinnane** [comment]
Changelog pls

**Comment by gumner** [comment]
i thought the changelog was broken or something because we had /tg/ changelog

File: code/modules/uplink/uplink_item.dm (hunk 2 above)
Comment (gumner) [reply code/modules/uplink/uplink_item.dm]: ok should work now

**Comment by KittyKathy** [comment]
You do it in your first post so instead of lol it would be something like this.
🆑 Gummer
add: Is that a JoJo reference?
tweak: Did someone say JoJo ops? kill me now.
/🆑

**Comment by gumner** [comment]
ehh i already did an .yml changelog

File: code/modules/uplink/uplink_item.dm (hunk 2 above)
Comment (KittyKathy) [reply code/modules/uplink/uplink_item.dm]: That's weird did you copy paste it or write it out? Probably better to copy it from another line so line 2229 where the surplus crate excludes from other modes than traitor you can just copy that line and paste it at line 567.

**Comment by yobob591** [comment]
sugg

File: code/modules/uplink/uplink_item.dm [hunk 3]
```diff
@@ -549,7 +549,7 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	desc = "A cyborg designed and programmed for systematic extermination of non-Syndicate personnel."
 	item = /obj/item/weapon/antag_spawner/nuke_ops/borg_tele
 	cost = 80
-
+	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/gang)
```
Comment (KittyKathy) [inline code/modules/uplink/uplink_item.dm]: Umm you put it at the wrong place that's for the Syndicate Cyborg correct line wrong place put it at line 567.

**Comment by JohnGinnane** [comment]
We're using a different way of doing changelogs. You do what Kitty said instead of making a changelog file, and a bot generates it

File: code/modules/uplink/uplink_item.dm (hunk 3 above)
Comment (gumner) [reply code/modules/uplink/uplink_item.dm]: kek im a tard

**Comment by gumner** [comment]
do i need to do it now that i did an .yml?

**Comment by JohnGinnane** [comment]
Yes, remove the .yml

**Comment by KittyKathy** [comment]
I don't think the yml changelogs work anymore just edit your first post and put this also delete the yml changelog file.

`:cl: Gumner`
`add: Ported standard holoparasite buffs`
`add: You can buy the holoparasite injector with traitor uplinks once again`
`/:cl:`

**Comment by KittyKathy** [comment]
Also don't forget to remove line 522 while makes the syndicate borg traitor only? meaning now nuke ops can't get it unless you fix that.

**Comment by gumner** [comment]
should work now

**Comment by ghost** [comment]
I'd rather just see holos back, I dunno how stronk the new /tg/ ones are?

**Comment by gumner** [comment]
they dont exist,basically

**Comment by ghost** [comment]
Guardian spirits do, I think. Saw something about them in the lavaland guide.

**Comment by gumner** [comment]
yeah tarot cards do exist in lavaland loot

**Comment by ghost** [comment]
By that I mean, bring back traitor kit and see how they fare before buffing them???

**Comment by gumner** [comment]
its exactly the same as they were before rebase,and standard was weaker on rebase because before rebase his damage was 30,and now its 20

**Comment by ghost** [comment]
No they weren't.
https://github.com/HippieStationCode/HippieStation13/blob/master/code/modules/mob/living/simple_animal/guardian/guardian.dm#L260
I'd also leave deflection out for now since I dunno how stronk the new holos are.

**Comment by yobob591** [comment]
they are just as strong as they were in the past dude

**Comment by gumner** [comment]
new holo's are complete shit and are not worth 20 tc

**Comment by gumner** [comment]
no holos were buffed(besides standard that im buffing rn)

**Comment by yobob591** [comment]
i completely forgot theres like 3 new holopara types that tg added

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 4]
```diff
@@ -409,6 +412,26 @@
 	update_action_buttons_icon()
 	return canmove
 
+/mob/living/simple_animal/bullet_act(obj/item/projectile/P)
```
Comment (j-mie) [inline code/modules/mob/living/simple_animal/simple_animal.dm:414]: Shouldn't this call `..()` somewhere?

Comment (gumner) [reply code/modules/mob/living/simple_animal/simple_animal.dm:414]: should work now

Comment (gumner) [reply code/modules/mob/living/simple_animal/simple_animal.dm:414]: should work now

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm [hunk 5]
```diff
@@ -10,6 +10,8 @@
 	tech_fluff_string = "<span class='holoparasite'>Boot sequence complete. Standard combat modules loaded. Holoparasite swarm online.</span>"
 	carp_fluff_string = "<span class='holoparasite'>CARP CARP CARP! You caught one! It's really boring and standard. Better punch some walls to ease the tension.</span>"
 	var/battlecry = "AT"
+	var/holopunchrng = null
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/types/standard.dm]: no need to initialize this as null

**Comment by ghost** [comment]
go do it

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by yobob591** [comment]
give me the jojoes

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm (hunk 5 above)
Comment (gumner) [reply code/modules/mob/living/simple_animal/guardian/types/standard.dm]: should work now

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm [hunk 6]
```diff
@@ -21,11 +23,25 @@
...
+				playsound(loc, src.attack_sound, 50, 1, 1)
+				playsound(loc, src.attack_sound, 50, 1, 1)
+				playsound(loc, src.attack_sound, 50, 1, 1)
+				playsound(loc, src.attack_sound, 50, 1, 1)
+				H.apply_effect(2, STUN)//change to 2 cause 1 does nothing apparently
+			if(20 to 50)
+				src.say("[src.battlecry]")
+				H.apply_effect(4, WEAKEN)
+			if(0 to 20)
+				if(istype(target, /mob/living/carbon/human))
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/types/standard.dm:42]: there is a macro for this

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm [hunk 7]
```diff
@@ -21,11 +23,25 @@
...
-		playsound(loc, src.attack_sound, 50, 1, 1)
-		playsound(loc, src.attack_sound, 50, 1, 1)
+/mob/living/simple_animal/hostile/guardian/punch/AttackingTarget(mob/living/T, mob/living/carbon/human/H)
 	..()
+	if(istype(T))
+		switch(holopunchrng)
+			if(50 to 100)
+				src.say("[src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry]\
+				[src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry]")
+				playsound(loc, src.attack_sound, 50, 1, 1)
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/types/standard.dm:33]: no spam please

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm [hunk 8]
```diff
@@ -21,11 +23,25 @@
...
-		playsound(loc, src.attack_sound, 50, 1, 1)
-		playsound(loc, src.attack_sound, 50, 1, 1)
-		playsound(loc, src.attack_sound, 50, 1, 1)
-		playsound(loc, src.attack_sound, 50, 1, 1)
+/mob/living/simple_animal/hostile/guardian/punch/AttackingTarget(mob/living/T, mob/living/carbon/human/H)
 	..()
+	if(istype(T))
+		switch(holopunchrng)
+			if(50 to 100)
+				src.say("[src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry]\
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/types/standard.dm:31]: no spam please

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by gumner** [comment]
it doesnt actually spam you normie,its just 3 characters being repeated

**Comment by WhiteFox08** [comment]
consider death

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm (hunk 6 above)
Comment (gumner) [reply code/modules/mob/living/simple_animal/guardian/types/standard.dm:42]: i dont get it

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm (hunk 8 above)
Comment (gumner) [reply code/modules/mob/living/simple_animal/guardian/types/standard.dm:31]: how the fuck is that spam

**Comment by gumner** [comment]
can someone fully review this

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 9]
```diff
@@ -78,6 +78,9 @@
 
 	var/attacked_sound = "punch" //Played when someone punches the creature
 
+	//bla bla bla reflect shit
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/simple_animal.dm:81]: This comment is bad

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm [hunk 10]
```diff
@@ -10,6 +10,8 @@
 	tech_fluff_string = "<span class='holoparasite'>Boot sequence complete. Standard combat modules loaded. Holoparasite swarm online.</span>"
 	carp_fluff_string = "<span class='holoparasite'>CARP CARP CARP! You caught one! It's really boring and standard. Better punch some walls to ease the tension.</span>"
 	var/battlecry = "AT"
+	var/holopunchrng
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/types/standard.dm:13]: This variable is never set

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm [hunk 11]
```diff
@@ -10,6 +10,8 @@
...
+	mob_reflect_chance = 40
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/types/standard.dm:14]: 40% chance on reflecting bullets? Seems a bit high

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm (hunk 8 above)
Comment (ghost) [reply code/modules/mob/living/simple_animal/guardian/types/standard.dm:31]: You are spamming the battlecry over and over for no reason, why?

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm (hunk 6 above)
Comment (ghost) [reply code/modules/mob/living/simple_animal/guardian/types/standard.dm:42]: There is a macro that checks if something is human, use it

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm [hunk 12]
```diff
@@ -21,11 +23,25 @@
...
+				playsound(loc, src.attack_sound, 50, 1, 1)
+				playsound(loc, src.attack_sound, 50, 1, 1)
+				playsound(loc, src.attack_sound, 50, 1, 1)
+				H.apply_effect(2, STUN)//change to 2 cause 1 does nothing apparently
+			if(20 to 50)
+				src.say("[src.battlecry]")
+				H.apply_effect(4, WEAKEN)
+			if(0 to 20)
+				if(istype(target, /mob/living/carbon/human))
+					src.say("[src.battlecry][src.battlecry][src.battlecry]")//i couldnt figure out how 2 do organ damage in this rebase so i just put other bad shit here.
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/guardian/types/standard.dm:43]: Bad comment

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 13]
```diff
@@ -408,6 +411,28 @@
 	update_action_buttons_icon()
 	return canmove
 
+/mob/living/simple_animal/bullet_act(obj/item/projectile/P)
+	if(!P)
+		return
+	if(prob(mob_reflect_chance))
+		visible_message("<span class='danger'>The [P.name] gets reflected by [name]!</span>")
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/simple_animal.dm:418]: I'm pretty sure there is a cleaner way to do this, try looking at reflective armor

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 9 above)
Comment (gumner) [reply code/modules/mob/living/simple_animal/simple_animal.dm:81]: fuck off thats grudge

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm (hunk 11 above)
Comment (gumner) [reply code/modules/mob/living/simple_animal/guardian/types/standard.dm:14]: only 10% more than ablative armor,and costs 20 tc

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 13 above)
Comment (gumner) [reply code/modules/mob/living/simple_animal/simple_animal.dm:418]: what you mean cleaner?

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm (hunk 10 above)
Comment (gumner) [reply code/modules/mob/living/simple_animal/guardian/types/standard.dm:13]: yeah i was told i dindt need to make it null some days ago

File: code/modules/mob/living/simple_animal/guardian/types/standard.dm (hunk 12 above)
Comment (gumner) [reply code/modules/mob/living/simple_animal/guardian/types/standard.dm:43]: dude,just a comment b r a h

**Comment by j-mie** [comment]
build this

**Comment by JohnGinnane** [comment]
@gumner you still working on this?

**Comment by ghost** [comment]
@JohnGinnane I think i scared him off with my review `:^)`

**Comment by gumner** [comment]
yes i will go back to this,i was busy w/ school

**Comment by gumner** [comment]
im closing this because i cant update my  branch for some reason

**Pull request closed without merging**